core/src/main/scala/org/apache/spark/scheduler/cluster/SparkDeploySchedulerBackend.scala

```diff
@@ -40,6 +40,14 @@ private[spark] class SparkDeploySchedulerBackend(
   var registrationDone = false
 
   val maxCores = conf.getOption("spark.cores.max").map(_.toInt)
+  val coreNumPerTask = {
+    val corePerTask = conf.getInt("spark.task.cpus", 1)
+    if (corePerTask < 1) {
+      throw new IllegalArgumentException(
+        s"spark.task.cpus is set to an invalid value $corePerTask")
+    }
+    corePerTask
+  }
```

Comment:
Is `corePerTask ` necessary? Why don't just assign `conf.getInt("spark.task.cpus", 1)` to `coreNumPerTask `?